### ISSUE TITLE
[spv-in] loop assignment error

### ISSUE THREAD

**Issue by Uriopass** [gfx-rs/naga#1025]
Naga succesfully parses and validates all my shaders except one, and I've used naga-cli to reduce it to the simplest form I could find.
Naga fails with
```
Function [1] '' is invalid:
        Expression [11] is invalid
        Used by a statement before it was introduced into the scope by any of the dominating blocks
```

On the following GLSL shader compiled using `glslc -O`:

```
#version 450

layout(location=0) out float out_f;

void main()
{
    float outer = 0;

    for (int i = 0; i < 1; i++) {
        float inner = 0.0;
        inner = 1.0;
        outer = inner;
    }

    out_f = outer;
}
```

Spir-V decompilation gives 
```
; SPIR-V
; Version: 1.0 
; Generator: Google Shaderc over Glslang; 8
; Bound: 33
; Schema: 0
               OpCapability Shader
          %1 = OpExtInstImport "GLSL.std.450"
               OpMemoryModel Logical GLSL450
               OpEntryPoint Fragment %4 "main" %29 
               OpExecutionMode %4 OriginUpperLeft
               OpDecorate %29 Location 0
       %void = OpTypeVoid
          %3 = OpTypeFunction %void
      %float = OpTypeFloat 32
    %float_0 = OpConstant %float 0
        %int = OpTypeInt 32 1
      %int_0 = OpConstant %int 0
      %int_1 = OpConstant %int 1
       %bool = OpTypeBool
    %float_1 = OpConstant %float 1
%_ptr_Output_float = OpTypePointer Output %float
         %29 = OpVariable %_ptr_Output_float Output
          %4 = OpFunction %void None %3
          %5 = OpLabel
               OpBranch %14 
         %14 = OpLabel
         %32 = OpPhi %float %float_0 %5 %float_1 %15 
         %31 = OpPhi %int %int_0 %5 %27 %15 
         %22 = OpSLessThan %bool %31 %int_1
               OpLoopMerge %16 %15 None
               OpBranchConditional %22 %15 %16 
         %15 = OpLabel
         %27 = OpIAdd %int %31 %int_1
               OpBranch %14 
         %16 = OpLabel
               OpStore %29 %32 
               OpReturn
               OpFunctionEnd
```

But I've attached the actual file for good measure.
[background_bug.frag.zip](https://github.com/gfx-rs/naga/files/6713717/background_bug.frag.zip)

**Comment by kvark** [comment]
cc @MatusT

**Comment by JCapucho** [comment]
Fixed by #1294
